src/cookie-io.js

```diff
@@ -114,6 +114,21 @@ class CharacterData {
         return this.characterData.clan.members;
     }
 
+    addFriend(name) {
+        if (!this.characterData.friends.includes(name)) {
+            this.characterData.friends.push(name);
+        }
+    }
+    removeFriend(name) {
+        const index = this.characterData.friends.indexOf(name);
+        if (index > -1) {
+            this.characterData.friends.splice(index, 1);
```

Comment:
TIL `splice`

Could also do `this.characterData.friends = this.characterData.friends.filter((friend) => friend != name);`
No preference either way